hphp/runtime/vm/jit/relocation-arm.cpp

```diff
@@ -126,367 +163,342 @@ InstrSet findLiterals(Instruction* start, Instruction* end) {
-          a.Mov(dst, instr->ImmPCOffsetTarget());
+  auto target = reinterpret_cast<TCA>(src->ImmPCOffsetTarget());
 
-          dest_count += (dest_block.frontier() - dest) / kInstructionSize;
-          is_relative = false;
-        }
-      } else if (instr->IsLoadLiteral()) {
-        if (!is_int19(imm) || far.count(instr)) {
-          dest_block.setFrontier(dest);
-          dest_count--;
-
-          vixl::MacroAssembler a { dest_block };
-          a.SetScratchRegisters(vixl::NoReg, vixl::NoReg);
-          auto const dst = vixl::Register(instr->Rd(), 64);
-          auto const tmp = dst.Is(rVixlScratch0)
-            ? rVixlScratch1 : rVixlScratch0;
-          a.Mov(tmp, instr->ImmPCOffsetTarget());
-          a.Ldr(dst, vixl::MemOperand(tmp));
-          a.SetScratchRegisters(rVixlScratch0, rVixlScratch1);
-
-          dest_count += (dest_block.frontier() - dest) / kInstructionSize;
-          is_relative = false;
-        }
-      } else if (instr->IsCondBranchImm()) {
-        imm >>= vixl::kInstructionSizeLog2;
-        if (!is_int19(imm) || far.count(instr)) {
-          dest_block.setFrontier(dest);
-          dest_count--;
-
-          vixl::MacroAssembler a { dest_block };
-          vixl::Label end;
-          auto const cond = static_cast<Condition>(instr->ConditionBranch());
-          auto const tmp = rVixlScratch0;
-          a.SetScratchRegisters(vixl::NoReg, vixl::NoReg);
-          a.B(&end, vixl::InvertCondition(cond));
-          a.Mov(tmp, instr->ImmPCOffsetTarget());
-          a.Br(tmp);
-          a.bind(&end);
-          a.SetScratchRegisters(rVixlScratch0, rVixlScratch1);
-
-          dest_count += (dest_block.frontier() - dest) / kInstructionSize;
-          is_relative = false;
-        }
-      } else if (instr->IsUncondBranchImm()) {
-        imm >>= vixl::kInstructionSizeLog2;
-        if (!is_int26(imm) || far.count(instr)) {
-          dest_block.setFrontier(dest);
-          dest_count--;
-
-          vixl::MacroAssembler a { dest_block };
-          a.SetScratchRegisters(vixl::NoReg, vixl::NoReg);
-          auto const tmp = rVixlScratch0;
-          a.Mov(tmp, instr->ImmPCOffsetTarget());
-          a.Br(tmp);
-          a.SetScratchRegisters(rVixlScratch0, rVixlScratch1);
-
-          dest_count += (dest_block.frontier() - dest) / kInstructionSize;
-          is_relative = false;
-        }
+  // If the target is outside of the range of this relocation,
+  // then update it.
+  if ((target < env.start) || (target >= env.end)) {
+    /*
+     * Calculate the new offset and determine if it can be encoded
+     * in a PC relative instruciton or if it needs to be converted
+     * to make use of an absolute target.
+     * Note: Use the VIXL scratch registers when transforming. Their
+     *       scope is just a single macroassembler directive, whereas
+     *       the scope of rAsm is an entire vasm instruction.
+     */
+    int imm = src->ImmPCOffsetTarget() - dest;
+    bool isRelative = true;
```

Comment:
This is a note rather than a nit, but: you shouldn't feel compelled to adhere to a particular case convention when writing new code.  Be consistent within a single file or subsystem (e.g., in `vasm-*` code, we tend towards `lower_case_with_underscores` everywhere), but outside of that, just do what's most natural for you.

(I only mention this because I see that you camelCase'd everything between your previous revision and this one.)